certification/pyxis/layers.go

```diff
@@ -2,68 +2,60 @@ package pyxis
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
-	"io"
 	"net/http"
-	"net/url"
-	"strings"
 
 	cranev1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/hasura/go-graphql-client"
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
 	log "github.com/sirupsen/logrus"
 )
 
 func (p *pyxisClient) CheckRedHatLayers(ctx context.Context, layerHashes []cranev1.Hash) ([]CertImage, error) {
-	layerIds := make([]string, 0, len(layerHashes))
+	layerIds := make([]graphql.String, 0, len(layerHashes))
 	for _, layer := range layerHashes {
-		layerIds = append(layerIds, layer.String())
+		layerIds = append(layerIds, graphql.String(layer.String()))
 	}
-	log.Tracef("the layerIds passed to pyxis are %s", layerIds)
 
-	pyxisQuery := url.QueryEscape(fmt.Sprintf("repositories.registry==registry.access.redhat.com and uncompressed_top_layer_id=in=(%s)", strings.Join(layerIds, ",")))
+	var query struct {
+		FindImages struct {
+			ContainerImage []struct {
+				UncompressedTopLayerId graphql.String `graphql:"uncompressed_top_layer_id"`
+				ID                     graphql.String `graphql:"_id"`
+			} `graphql:"data"`
+			Error struct {
+				Status graphql.Int    `graphql:"status"`
+				Detail graphql.String `graphql:"detail"`
+			} `graphql:"error"`
+			Total graphql.Int
+			Page  graphql.Int
+		} `graphql:"find_images(filter: {and:[{repositories:{registry:{in:$registries}}}{uncompressed_top_layer_id:{in:$contImageLayers}}]})"`
+	}
 
-	req, err := p.newRequest(
-		ctx,
-		http.MethodGet,
-		fmt.Sprintf("%s?filter=%s", p.getPyxisUrl("images"), pyxisQuery),
-		nil,
-	)
-	if err != nil {
-		log.Error(err)
-		return nil, err
+	variables := map[string]interface{}{
+		"contImageLayers": layerIds,
+		"registries":      []graphql.String{"registry.access.redhat.com"},
 	}
 
-	log.Tracef("URL is %s", req.URL)
-	resp, err := p.Client.Do(req)
-	if err != nil {
-		log.Error(err)
-		return nil, err
+	httpClient, ok := p.Client.(*http.Client)
```

Comment:
This is the Client that's on the `pyxisClient` struct correct? Or is it a net new HTTP Client?

Reply:
That's what I thought but I wanted to make sure, since the other code where that is instantiated isn't changing was hard to tell.